app/assets/src/components/common/DetailsSidebar/GeneDetailsMode/GeneDetailsMode.jsx

```diff
@@ -0,0 +1,430 @@
+        }
+        default: {
+          break;
+        }
+      }
+    });
+
+    return geneInfo;
+  }
+
+  //*** Callback functions ***
+
+  expandOntology = () => {
+    this.setState({ collapseOntology: false });
+  };
+
+  //*** Functions depending on state ***
+
+  generateLinkTo(source) {
+    const { ontology } = this.state;
+    const { geneName } = this.props;
+    switch (source) {
+      case SOURCE_CARD: {
+        return URL_CARD_ARO + ontology.accession;
+      }
+      case SOURCE_PUBMED: {
+        return URL_PUBMED + "?term=" + geneName;
+      }
+      case SOURCE_GOOGLE_SCHOLAR: {
+        return URL_GOOGLE_SCHOLAR + geneName;
+      }
+      case SOURCE_NCBI_REF_GENE: {
+        return URL_NCBI_REF_GENE + geneName;
+      }
+      default: {
+        return "";
+      }
+    }
+  }
+
+  //*** Render methods ***
+
+  renderCARDLicense() {
+    const { ontology } = this.state;
+    return (
+      <div className={cs.cardLicense}>
+        This article uses material from the CARD Antibiotic Resistance Ontology
+        entry for{" "}
+        <a
+          href={this.generateLinkTo(SOURCE_CARD)}
+          className={cs.cardLink}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          {ontology.label}
+        </a>, which is released under the{" "}
+        <a
+          href="https://creativecommons.org/licenses/by/4.0/"
+          className={cs.cardLink}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Creative Commons CC-BY license version 4.0
+        </a>{" "}
+        by McMaster University.
+      </div>
+    );
+  }
+
+  renderNotFound() {
+    const { geneName } = this.props;
+    return <div className={cs.cardLicense}>No data found for {geneName}</div>;
+  }
+
+  renderHeader() {
+    const { loading, ontology } = this.state;
+    const { geneName } = this.props;
+    if (loading) {
+      return <div className={cs.loadingMsg}>Loading...</div>;
+    }
+    return (
+      <div className={cs.title}>
+        {ontology.label !== "---" ? ontology.label : geneName}
+      </div>
+    );
+  }
+
+  renderOntology() {
+    const { ontology, collapseOntology } = this.state;
+    return (
+      <div>
+        <div className={cs.text}>
+          <div className={cs.textInner}>
+            {CARD_SYNONYMS}: <em>{ontology.synonyms}</em>
+          </div>
+        </div>
+        <div className={cs.subtitle}>Description</div>
+        <div className={cs.text}>
+          <div className={cs.textInner}>{ontology.description}</div>
+        </div>
+        {!collapseOntology && (
+          <div>
+            <div className={cs.subtitle}>{CARD_FAMILY}</div>
+            <div className={cs.text}>
+              <div className={cs.textInner}>{ontology.geneFamily}</div>
+            </div>
+            <div className={cs.subtitle}>{CARD_CLASS}</div>
+            <div className={cs.text}>
+              <div className={cs.textInner}>{ontology.drugClass}</div>
```

Comment:
It might look nicer to have the first letter capitalized in the drug class and mechanisms

Reply:
Done!